syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionIrcConnectorJSONLoadingTest.java

```diff
@@ -0,0 +1,180 @@
+		assertThat(extension.getDependencies()).filteredOn("id", "io.syndesis.integration:integration-runtime:jar:1.3-SNAPSHOT").isNotNull(); 
+		assertThat(extension.getDependencies()).filteredOn("id", "org.apache.camel:camel-core:jar:2.20.1").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.apache.camel:camel-irc:jar:2.20.1").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.apache.camel:camel-spring-boot:jar:2.20.1").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.springframework.boot:spring-boot-autoconfigure:jar:1.5.8.RELEASE").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.springframework.boot:spring-boot-loader:jar:1.5.8.RELEASE").isNotNull();
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getExtensionType()}.
+	 */
+	@Test
+	public void testGetExtensionType() {
+		assertEquals("extension type value is not matching", "Connectors", extension.getExtensionType());
+	}
+		
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getActions()}.
+	 */
+	@Test
+	public void testGetActions() {
+		assertTrue("actions are not correct", extension.getActions() != null && !extension.getActions().isEmpty() && extension.getActions().size() == 1);
+	}
+
+	@Test
+	public void testSyndesisAction() {
+		SyndesisAction action = extension.getActions().get(0);
+		
+		assertEquals("Action ID not matching", "io.syndesis:irc-privmsg", action.getId());
+
+		assertEquals("Action Name not matching", "IRC PRIVMSG", action.getName());
+
+		assertEquals("Action Pattern not matching", "From", action.getPattern());
+		
+		assertEquals("Action Type not matching", "connector", action.getActionType());
+
+		assertEquals("Action Description not matching", "React to privmsg", action.getDescription());
+	}
+
+	@Test
+	public void testSyndesisActionDescriptor() {
+		SyndesisActionDescriptor descriptor = extension.getActions().get(0).getDescriptor();
+		
+		ActionDataShape inputShape = descriptor.getInputDataShape();
+		assertThat(inputShape.getKind()).isEqualTo("any");
+
+		ActionDataShape outputShape = descriptor.getOutputDataShape();
+		assertThat(outputShape.getKind()).isEqualTo("any");
+	}
+
+	@Test
+	public void testPropertyDefinitionSteps() {
+		List<PropertyDefinitionStep> steps = extension.getActions().get(0).getDescriptor().getPropertyDefinitionSteps();
+		assertThat(steps).hasSize(1);
+		
+		PropertyDefinitionStep step = steps.get(0);
+		assertThat(step.getDescription()).isEqualTo("Properties");
+		assertThat(step.getName()).isEqualTo("properties");
+		assertThat(step.getProperties()).hasSize(4);
+	}
+
+	@Test
+	public void testStepProperties() {
+		Map<String, SyndesisExtensionProperty> properties = extension.getActions().get(0).getDescriptor().getPropertyDefinitionSteps().get(0).getProperties();
+		assertThat(properties.keySet()).contains(
```

Comment:
you can use directly
`		assertThat(properties).containsKeys(
				"nickname",
				"hostname",
				"port",
				"channels"
		);`

Reply:
less code is better